Deal with space after +/-

billdueber opened the pull request:
...when doing solr escaping.

Note this does _not_ yet test if actually works with solr; coming next.

billdueber committed:
Deal with space after +/-

antmoth commented:
All the backslashes making my head hurt, but if the tests pass this seems like a reasonable start?

Commit:
Lift errors and warnings up from mini-search onto search

We may want to add to errors and warnings at the Search object level,
not just from the MiniSearch pre-processing

Commit:
Update 00-catalog.yml

Commit:
scriptlet to copy files from the k8s config

Commit:
broke my own script. fixed it.

billdueber committed:
Symbolize keys in solr parameters

Commit:
make it easier to get the wanted tokens string to spectrum

Commit:
Should only need clean string

bertrama committed:
Send details upstream

Send a detailed message upstream about the nature of the parse error,
the original search string and the actual search string.

bertrama committed:
Add a to_h method.

antmoth commented on lib/mlibrary_search_parser/search_handler.rb:
```diff
@@ -2,13 +2,51 @@
 module MLibrarySearchParser
-  class UnevenParensError < RuntimeError; end
+  class ParseError
+    DETIALS = "ParseError"
```
probably needs to be DETAILS

antmoth commented on lib/mlibrary_search_parser/transform/solr/local_params.rb:
```diff
@@ -13,6 +13,7 @@ def transform!
           else
             super
             set_param("q", "_query_:#{query}")
+            set_param("clean_string", original_search_tree.clean_string)
```
I think we don't need this any more

billdueber replied:
Changing it to a method on solr transforms so you can do `localparams_obj.clean_string`, parallel with what we do with everything else.

antmoth commented on spec/spec_helper.rb:
```diff
@@ -45,6 +45,8 @@ def not_node(node_or_string)
 @config_file = './spec/data/00-catalog.yml'
 @config = YAML.load(ERB.new(File.read(@config_file)).result)
 TEST_HANDLER =  MLibrarySearchParser::SearchHandler.new(@config)
+SEARCH_FACTORY =  MLibrarySearchParser::Search.search_builder(@config)
```
is this being used?

antmoth reviewed (commented):
I flagged a couple small things that aren't actually related to the lucene escaping but overall this looks fine

billdueber committed:
Minor fixes:
  * Typo on word 'DETIALS'
  * Remove unused stuff from spec_helper
  * Remove localparams.params["clean_string"] in favor of
    adding a `clean_string` method to the solr transforms

Commit:
Merge master

antmoth committed:
Merge branch 'master' into SEARCH-1271_escape_plus_minus_space
